runtime/src/main/java/org/corfudb/protocols/wireprotocol/IMetadata.java

```diff
@@ -131,12 +131,12 @@ default Long getThreadId() {
      */
     @SuppressWarnings("unchecked")
     default Long getGlobalAddress() {
-        if (getMetadataMap() == null
-                || getMetadataMap().get(LogUnitMetadataType.GLOBAL_ADDRESS) == null) {
-            return -1L;
+        if (getMetadataMap() == null) {
+            return Address.NON_ADDRESS;
```

Comment:
I don't think you need this, Its either empyt, or it doesn't have the key.

Reply:
Done.